File: test_regress/t/t_var_escape.pl
#!/usr/bin/env perl
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2003-2009 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

scenarios(simulator => 1);

compile(
    # Access is so we can dump waves
    v_flags2 => [$Self->{vlt_all} ? '-trace' : ' +access+rwc'],
    );

execute(
    check_finished => 1,
    );

if ($Self->{vlt_all}) {
    file_grep     ("$Self->{obj_dir}/simx.vcd", qr/\$enddefinitions/x);
    my $sig = quotemeta("bra[ket]slash/dash-colon:9");
    file_grep     ("$Self->{obj_dir}/simx.vcd", qr/ $sig/);
    file_grep     ("$Self->{obj_dir}/simx.vcd", qr/ other\.cyc /);
    file_grep     ("$Self->{obj_dir}/simx.vcd", qr/ module mod\.with_dot /);
    vcd_identical ("$Self->{obj_dir}/simx.vcd",
                   $Self->{golden_filename});
}

ok(1);
1;

File: test_regress/t/t_var_escape.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2024 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('simulator')

test.compile(
    # Access is so we can dump waves
    v_flags2=['-trace' if test.vlt_all else ' +access+rwc'])

test.execute()

if test.vlt_all:
    test.file_grep(test.trace_filename, r'\$enddefinitions')
    sigre = re.escape("bra[ket]slash/dash-colon:9")
    test.file_grep(test.trace_filename, sigre)
    test.file_grep(test.trace_filename, r' other\.cyc ')
    test.file_grep(test.trace_filename, r' module mod\.with_dot ')
    test.vcd_identical(test.trace_filename, test.golden_filename)

test.passes()
